### PR TITLE
Bugfix in YarpRobotControl::Impl::setReferences

### PR DESCRIPTION
**Pull request by GiulioRomualdi**
This fixes https://github.com/dic-iit/bipedal-locomotion-framework/issues/270

### CODE REVIEW

File: src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp [hunk 1]
```diff
@@ -451,20 +457,26 @@ struct YarpRobotControl::Impl
...
                     const auto jointError = std::abs(jointValues[indeces[i]]
                                                      - this->positionFeedback[indeces[i]]);
 
                     constexpr double scaling = 180 / M_PI;
                     constexpr double maxVelocityInDegPerSeconds = 3.0;
-                    refSpeeds[i] = std::max(maxVelocityInDegPerSeconds,
-                                            scaling * (jointError / this->positioningDuration));
+                    this->positionControlRefSpeeds[i]
+                        = std::max(maxVelocityInDegPerSeconds,
+                                   scaling * (jointError / this->positioningDuration));
```
Comment (traversaro) [inline src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp:469]: I am not sure of this logic, but this is not part of what is changed by the PR.

**Review by traversaro — APPROVED** [review]

File: src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp [hunk 2]
```diff
@@ -304,7 +309,7 @@ struct YarpRobotControl::Impl
         this->controlModes.resize(this->actuatedDOFs);
         this->controlModesYarp.resize(this->actuatedDOFs);
         this->axesName.resize(this->actuatedDOFs);
-
+        this->positionControlRefSpeeds.resize(this->actuatedDOFs);
```
Comment (S-Dafarra) [inline src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp:312]: I am just checking the modification, so I might be missing something, but it seems that you are resizing this vector in two different places, in two different ways. Above I see:
```
this->positionControlRefSpeeds.resize(
        this->desiredJointValuesAndMode.index[IRobotControl::ControlMode::Position].size());
```

File: src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp (hunk 1 above)
Comment (S-Dafarra) [reply src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp:469]: Actually, ``maxVelocityInDegPerSeconds`` is misleading, as it is instead the minimum velocity. Each joint is supposed to reach the desired position in the time specified by ``positioningDuration``. If the joint is already close to the desired configuration, we don't want the joint to go too slow, so we set the desired velocity to ``maxVelocityInDegPerSeconds``, that should be called ``minVelocityInDegPerSeconds``.

**Review by S-Dafarra — APPROVED** [review]

File: src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp (hunk 1 above)
Comment (GiulioRomualdi) [reply src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp:469]: Thank you @S-Dafarra I reneame `maxVelocityInDegPerSeconds` into `minVelocityInDegPerSeconds`

File: src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp (hunk 2 above)
Comment (GiulioRomualdi) [reply src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp:312]: Hi @S-Dafarra, both the resize are required. 

Indeed when [`setDriver()`](https://github.com/dic-iit/bipedal-locomotion-framework/blob/317b86fffcbcd2f263bf76785c11269df5aa3ef7/src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp#L244) is called the motor control mode for each joint is retrieved. 

https://github.com/dic-iit/bipedal-locomotion-framework/blob/317b86fffcbcd2f263bf76785c11269df5aa3ef7/src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp#L323

Suppose that all the joints are in position when you call

https://github.com/dic-iit/bipedal-locomotion-framework/blob/317b86fffcbcd2f263bf76785c11269df5aa3ef7/src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp#L586

You will have a segfault if this resize is missing:

https://github.com/dic-iit/bipedal-locomotion-framework/blob/317b86fffcbcd2f263bf76785c11269df5aa3ef7/src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp#L312

On the other hand 

https://github.com/dic-iit/bipedal-locomotion-framework/blob/317b86fffcbcd2f263bf76785c11269df5aa3ef7/src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp#L208-L209

is required when you ask for a different control mode. (Now the number of joints in position mode may be different) 

---

By the way, I think it is more clear to change the resize as 
```suggestion
        this->positionControlRefSpeeds.resize(this->desiredJointValuesAndMode.index[IRobotControl::ControlMode::Position].size());
```
and call it after [this line](https://github.com/dic-iit/bipedal-locomotion-framework/blob/317b86fffcbcd2f263bf76785c11269df5aa3ef7/src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp#L346).

**Pull request merged**